packages/block-editor/src/hooks/spacing.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+import { Platform } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PaddingEdit,
+	hasPaddingSupport,
+	useIsPaddingDisabled,
+} from './padding';
+import SpacingPanelControl from '../components/spacing-panel-control';
+
+export const SPACING_SUPPORT_KEY = 'spacing';
+
+/**
+ * Inspector controls for spacing support.
+ *
+ * @param  {Object} props Block props.
+ * @return {WPElement}    Inspector controls for spacing support features.
+ */
+export function SpacingPanel( props ) {
+	const isDisabled = useIsSpacingDisabled( props );
+	const isSupported = hasSpacingSupport( props.name );
+
+	if ( isDisabled || ! isSupported ) {
+		return null;
+	}
+
+	return (
+		<SpacingPanelControl key="spacing">
```

Comment:
What would you think of inlining `SpacingPanelControl`? This is: remove that component and use the sub-components directly in here, as we do in the typography hook et all. This has some code duplication that's difficult to reason about (we check for isDisabled and isSpacingEnabled in two places: here and within `SpacingPanelControl`).

Reply:
Definitely an improvement. Done ✅ 

It was a remnant from when I originally set out to later reuse the panel in the global styles sidebar.